ballsdex/packages/players/cog.py

```diff
@@ -449,9 +452,48 @@ async def give(
         countryball.favorite = False
         await countryball.save()
 
-        await interaction.response.send_message(
+        await interaction.followup.send(
             f"You just gave the {settings.collectible_name} "
             f"{countryball.description(short=True, include_emoji=True, bot=self.bot)} to "
             f"{user.mention}!"
         )
         del self.bot.locked_balls[countryball.id]
+
+    @balls.command(name="count")
+    @app_commands.checks.has_any_role(*settings.root_role_ids)
+    async def balls_count(
+        self,
+        interaction: discord.Interaction,
+        ball: BallEnabledTransform = None,
+        shiny: bool = None,
+        special: SpecialEnabledTransform = None,
+    ):
+        """
+        Count the number of balls that you have.
+
+        Parameters
+        ----------
+
+        ball: Ball
+            The ball you want to count.
+        shiny: bool
+            Whether the ball is shiny or not.
+        special: Special
+            The special background of the ball.
+        """
+        filters = {}
+        if ball:
+            filters["ball"] = ball
+        if shiny is not None:
+            filters["shiny"] = shiny
+        if special:
+            filters["special"] = special
+
+        await interaction.response.defer(ephemeral=True, thinking=True)
+        balls = await BallInstance.filter(**filters)
+        country = f"{ball.country} " if ball else ""
+        plural = "s" if len(balls) > 1 else ""
+        special = f"{special.name} " if special else ""
+        await interaction.followup.send(
+            f"You have {len(balls)} {special}{country}{settings.collectible_name}{plural}."
```

Comment:
No need to query all colums
```suggestion
        balls = await BallInstance.filter(**filters).count()
        country = f"{ball.country} " if ball else ""
        plural = "s" if balls > 1 else ""
        special = f"{special.name} " if special else ""
        await interaction.followup.send(
            f"You have {balls} {special}{country}{settings.collectible_name}{plural}."
```